### DISCUSSION TITLE
VS Code

### DISCUSSION THREAD

**Discussion by ghost** [pnp/powershell#994] [Q&A]
When using PnP.Powershell in VS Code with PowerShell extention I see lots of errors making it more or less unusable.

There are no errors in normal Windows PowerShell or ISE PowerShell.

Anyone else experiencing this, and is there any workaround?

**Comment by AndersRask** [comment]
Hi,

you need to be a bit more specific if you want an answer :)

What errors exactly? Can you provide a screenshot?

Please give os something to work with...

kind regards
Anders Rask

**Comment by ghost** [comment]
Sorry, here you go :)

These are just an example of errors, that only show in VS code PS and not in Windows PS.

Add-PnPTenantTheme : Method not found: 'Int32 System.Text.Encodings.Web.TextEncoder.FindFirstCharacterToEncodeUtf8(System.ReadOnlySpan`1<Byte>)'

Invoke-PnPSiteTemplate : Method not found: 'System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1<Boolean> Enumerator.MoveNextAsync()'

**Reply by AndersRask** [reply]
This could be version dependent, as you may use 5.1 for your Windows PowerShell, but out of box VS Code uses PS 7.

Check this by typing $PSVersionTable

If the versions differ, the solution is to install the module for both versions, as the modules reside in different locations.

You can see the individual module path locations by typing $env:psmodulepath in the individual consoles.

` $env:PSModulePath -split ";"`

On Windows 10 the one used for install is usually
PS 5.1:  C:\Program Files\WindowsPowerShell\Modules
PS 7.2: c:\program files\powershell\7\Modules

**Reply by ghost** [reply]
Good point, but unfortunately already checked earlier. Any other ideas? :)

Below info:

$PSVersionTable
Name                           Value
----                           -----
PSVersion                      5.1.19041.1023
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.19041.1023
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1

C:\Program Files\WindowsPowerShell\Modules <----- Contains latest version per today and only latest version (ie. C:\Program Files\WindowsPowerShell\Modules\PnP.PowerShell\1.7.0)

C:\Users\jmj\OneDrive - Timengo\Dokumenter\WindowsPowerShell\Modules <----- Empty

c:\program files\powershell\7\Modules <----- Does not exist

**Reply by AndersRask** [reply]
did you run the commands from inside the VS Code shell as well?

**Reply by ghost** [reply]
That works, as this is just same shell as Windows PS.

The issue is when using the PowerShell extention available in VS Code (https://github.com/PowerShell/vscode-powershell).

It is probably an issue with the extention, but maybe there is a workaround....guess not?

**Reply by AndersRask** [reply]
It looks like VSCode cannot find your modules.
Can you see the module from within VSCode shell?

`Get-Module -ListAvailable *pnp*`

Does other modules work from within VSCode?

**Reply by ghost** [reply]
I do not have any issues in other modules (AzureADPreview, Microsoft Teams and several others) - only using pnp.powershell I have issues. Some parts of Pnp.PowerShell works fine and some does not. Really strange.

**Comment by TobyLeduc** [comment]
> When using PnP.Powershell in VS Code with PowerShell extention I see lots of errors making it more or less unusable.
> 
> There are no errors in normal Windows PowerShell or ISE PowerShell.
> 
> Anyone else experiencing this, and is there any workaround?

There is definately something going on with the vscode powershell integrated console terminal with versions 1.6.0 and 1.7.0. There is some conversation going on here https://github.com/PowerShell/vscode-powershell/issues/3510. It appears that there are no issues when running standard powershell, powershell ISE or the powershell terminal in vscode.
